File: java-api/src/main/java/de/davelee/personalman/api/AddTimeSheetHoursRequest.java
package de.davelee.personalman.api;

import lombok.*;

/**
 * This class is part of the PersonalMan REST API. It represents a request to add a number of hours to a particular date
 * to the user's timesheet.
 * @author Dave Lee
 */
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class AddTimeSheetHoursRequest {

    //company that the person works for
    private String company;

    //username of the person
    private String username;

    //token of the user making the change
    private String token;

    //the date to add the hours to in format dd-MM-yyyy.
    private String date;

    //the number of hours to add
    private int hours;

}

File: server/src/users/requests/addtimesheethours.request.ts
import { ApiProperty } from '@nestjs/swagger';

export class AddTimesheetHoursRequest {

    //company that the person works for
    @ApiProperty()
    private company: string;

    //username of the person
    @ApiProperty()
    private username: string;

    //token of the user making the change
    @ApiProperty()
    private token: string;

    //the date to add the hours to in format dd-MM-yyyy.
    @ApiProperty()
    private date: string;

    //the number of hours to add
    @ApiProperty()
    private hours: number;

    getCompany(): string {
        return this.company;
    }

    getUsername(): string {
        return this.username;
    }

    getToken(): string {
        return this.token;
    }

    getHours(): number {
        return this.hours;
    }

    getDate(): string {
        return this.date;
    }

}